### PR TITLE
`General`: Fix active students chart for empty courses

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
The Server had thrown an exception when the course started later than in 9 Weeks.

### Description
This PR adds some fallbacks, and streamlines some unnecessary code.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with StartDate & EndDate at least 9 weeks in the future (e.g. https://artemis-test4.artemis.cit.tum.de/course-management/21)
- 1 Course with no StartDate & EndDate set (e.g. https://artemis-test4.artemis.cit.tum.de/course-management/22)
- 1 active Course with dates set & student data (e.g. https://artemis-test4.artemis.cit.tum.de/course-management/3)

1. Log in to Artemis
2. Navigate the course detail page of these courses 
3. Verify that all the buttons work as intended
image

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
image

## Summary by CodeRabbit

- **New Features**
  - Courses now allow specification of the statistics period size for fetching data.

- **Bug Fixes**
  - Adjusted student activity slicing to ensure correct data size in course detail view.

- **Refactor**
  - Improved conditional rendering in course detail line chart for enhanced performance.

- **Tests**
  - Updated course management tests to include new `periodSize` parameter and verify its impact on student activity data.

### CODE REVIEW

**Review by laurenzfb — APPROVED** [review]
tested on ts4, works fine. code lgtm

**Review by farisd16 — APPROVED** [review]
Works as expected on ts4

File: src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java [hunk 1]
```diff
@@ -2952,14 +2951,20 @@ public void testGetCourseManagementDetailData() throws Exception {
 
         // Active Users
         int periodIndex = 0;
+        int periodSize = 8;
         LinkedMultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
         parameters.add("periodIndex", Integer.toString(periodIndex));
+        parameters.add("periodSize", Integer.toString(periodSize));
 
-        var activeStudents = request.get("/api/courses/" + course1.getId() + "/statistics", HttpStatus.OK, Integer[].class, parameters);
+        Integer[] activeStudents = request.get("/api/courses/" + course1.getId() + "/statistics", HttpStatus.OK, Integer[].class, parameters);
```
Comment (reschandreas) [inline src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java:2959]: can't you leave it as `var`?

**Review by reschandreas — APPROVED** [review]
tested on ts4, works as expected, code lgtm. thanks for directly providing courses 🚀

**Review by xHadie — DISMISSED** [review]
_Manually tested on TS11_Legacy_
Everything works as expected for all course start/end date scenarios

**Review by laurenzfb — APPROVED** [review]

**Review by xHadie — APPROVED** [review]

**Review by egekurt123 — APPROVED** [review]
Tested on ts4, works as expected

**Review by konrad2002 — APPROVED** [review]
manually tested on ts4, works as expected

**Review by LenaKahle — APPROVED** [review]
Approve, tested on ts4.

**Pull request merged**